src/registrar/admin.py

```diff
@@ -716,16 +719,58 @@ def response_change(self, request, obj):
         return super().response_change(request, obj)
 
     def do_delete_domain(self, request, obj):
+        if not isinstance(obj, Domain):
+            # Could be problematic if the type is similar,
+            # but not the same (same field/func names) so we err out.
+            # We do not want to accidentally delete records.
+            raise ValueError("Object is not of type Domain")
```

Comment:
what happens on the admin page when you throw this will it be a 500 error?

Reply:
@abroddrick Correct. This was intentional, if this happens then it indicates something downstream went very wrong and would warrant investigation. Can remove the check or just do a normal message though

Reply:
I would say, have it be a normal error message. We are trying to avoid the 500 errors where possible and that seems easy here since you can copy code